chainer/functions/activation/log_softmax.py

```diff
@@ -79,13 +95,29 @@ def backward(self, x, gy):
             desc = cudnn.create_tensor_descriptor(gx_cube)
             libcudnn.softmaxBackward(
                 handle, _algorithm, _mode, one.data, desc.value,
-                y.data.ptr, desc.value, gy[0].data.ptr, zero.data,
+                y.data.ptr, desc.value, gy.data.ptr, zero.data,
                 desc.value, gx.data.ptr)
         else:
-            gx = gy[0] - xp.exp(y) * gy[0].sum(axis=1, keepdims=True)
-
+            gx = gy - xp.exp(y) * gy.sum(axis=1, keepdims=True)
         return gx,
 
+    def backward(self, indexes, ggx):
+        y, gy = self.get_retained_inputs()
+        ret = []
+        exp_y = exponential.exp(y)
+        if 0 in indexes:
+            gy_sum = sum_.sum(gy, 1, True)
+            gy_sum = broadcast.broadcast_to(gy_sum, gy.shape)
+            g0 = -ggx[0] * exp_y * gy_sum
+            ret.append(g0)
+        if 1 in indexes:
+            # TODO(Kenta Oono): implement F.dot
```

Comment:
I thought this computation cannot be implemented with `dot`. We may use `F.matmul` by appropriately inserting singleton dimensions.